### PR TITLE
Stix export upgrade

### PR DESCRIPTION
**Pull request by Ni-Knight**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Upgrades the STIX export function allowing to export almost all of the XSAOR indicator types to STIX format without breaking BC (SCOs are still exported as Indicator SDO type).

AHA - https://xsoar.ideas.aha.io/ideas/FR-I-2525

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by Ni-Knight** [comment]
@yaakovi - all green, need some CR ❤️

File: Packs/CommonScripts/Scripts/StixCreator/StixCreator.py [hunk 1]
```diff
@@ -3,125 +3,157 @@
...
+        "attack pattern": AttackPattern,
+        "campaign": Campaign,
+        "infrastructure": Infrastructure,
+        "tool": Tool,
+        "intrusion set": IntrusionSet,
+        "report": Report,
+        "threat actor": ThreatActor,
+        "cve": Vulnerability,
+        "course of action": CourseOfAction
+    }
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/StixCreator/StixCreator.py:45]: move outside of main

File: Packs/CommonScripts/Scripts/StixCreator/StixCreator.py [hunk 2]
```diff
@@ -3,125 +3,157 @@
...
-                   raw_response=str(indicators[0]))
-else:
-    context = {
-        'StixExportedIndicators': {}
-    }
-    return_outputs(readable_output="",
-                   outputs=context,
-                   raw_response={})
+            indicator_type = demisto_indicator_type.lower().replace("-", "")
+            indicator = Indicator(pattern=SCOs[indicator_type].format(value),
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/StixCreator/StixCreator.py:96]:
```suggestion
            indicator = Indicator(pattern=f"[{SCOs[indicator_type]} = '{value}']",
```

File: Packs/CommonScripts/Scripts/StixCreator/StixCreator.py [hunk 3]
```diff
@@ -3,125 +3,157 @@
...
 
 from stix2 import Bundle, ExternalReference, Indicator, Vulnerability
+from stix2 import AttackPattern, Campaign, Malware, Infrastructure, IntrusionSet, Report, ThreatActor
+from stix2 import Tool, CourseOfAction
+from typing import Any, Callable
+
+
+def main():
+    SCOs: dict[str, str] = {
+        "file md5": "[file:hashes.md5 ='{}']",
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/StixCreator/StixCreator.py:18]: same for all values in dict (btw, you missed {} in line 21)
```suggestion
        'file md5': 'file:hashes.md5',
```

File: Packs/CommonScripts/Scripts/StixCreator/StixCreator.py [hunk 4]
```diff
@@ -3,125 +3,157 @@
...
-                   raw_response={})
+            indicator_type = demisto_indicator_type.lower().replace("-", "")
+            indicator = Indicator(pattern=SCOs[indicator_type].format(value),
+                                  pattern_type='stix',
+                                  **kwargs)
+
+            indicators.append(indicator)
+
+        except KeyError:
+            try:
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/StixCreator/StixCreator.py:103]: add a log message for future debugging

File: Packs/CommonScripts/Scripts/StixCreator/StixCreator.py (hunk 3 above)
Comment (Ni-Knight) [reply Packs/CommonScripts/Scripts/StixCreator/StixCreator.py:18]: done

File: Packs/CommonScripts/Scripts/StixCreator/StixCreator.py (hunk 1 above)
Comment (Ni-Knight) [reply Packs/CommonScripts/Scripts/StixCreator/StixCreator.py:45]: done

File: Packs/CommonScripts/Scripts/StixCreator/StixCreator.py (hunk 4 above)
Comment (Ni-Knight) [reply Packs/CommonScripts/Scripts/StixCreator/StixCreator.py:103]: done

File: Packs/CommonScripts/ReleaseNotes/1_7_13.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#### Scripts
+##### StixCreator
+You can now export almost all of XSOAR indicator types to STIX objects.
```
Comment (yaakovi) [inline Packs/CommonScripts/ReleaseNotes/1_7_13.md:3]:
```suggestion
- You can now export almost all of the XSOAR indicator types to STIX objects.
- Updated the Docker image to: demisto/py3-tools:1.0.0.31193
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**